cli/diag.c

```diff
@@ -1911,6 +1923,13 @@ static int pattern(int argc, char **argv)
 			switchtec_perror("pattern_gen_set");
 			return -1;
 		}
+		if (cfg.disable)
+			printf("Disabled pattern generator on port %d\n",
+				cfg.port_id);
+		else
+			printf("Pattern generator set for port %d with pattern type %s at %s\n", 
+				cfg.port_id, pattern_to_str(cfg.pattern), 
+				link_speed_to_str(cfg.link_speed));
```

Comment:
This commit is doing a lot of things and probably can be break down into atomic ones.